app/src/main/java/com/duycuong/weather/data/source/remote/WeatherRemoteDataSource.java

```diff
@@ -0,0 +1,124 @@
+                        .append(Constant.FORECAST_PART)
+                        .append("/")
+                        .append(BuildConfig.DASK_SKY_API_KEY)
+                        .append("/")
+                        .append(location.getLatitude() + "")
+                        .append(",")
+                        .append(location.getLongitude())
+                        .append("?")
+                        .append(Constant.EXCLUDE_PART)
+                        .append("=")
+                        .append(Constant.MINUTELY_PART)
+                        .append(",")
+                        .append(Constant.HOURLY_PART)
+                        .append(",")
+                        .append(Constant.ALERTS_PART)
+                        .append(",")
+                        .append(Constant.FLAGS_PART)
+                        .append(",")
+                        .append(Constant.DAILY_PART);
+
+        new LoadDataAsync(new Callback<String>() {
+            @Override
+            public void onStartLoading() {
+                callback.onStartLoading();
+            }
+
+            @Override
+            public void onGetSuccess(String data) {
+                try {
+                    if (data == null) {
+                        callback.onGetFailure(
+                                Resources.getSystem().getString(R.string.title_load_failure));
+                        return;
+                    }
+
+                    Gson gson = new Gson();
+                    callback.onGetSuccess(gson.fromJson(data, CurrentlyResponse.class));
+                } catch (JsonParseException e) {
+                    callback.onGetFailure(e.getMessage());
+                }
+            }
+
+            @Override
+            public void onGetFailure(String message) {
+                callback.onGetFailure(message);
+            }
+
+            @Override
+            public void onComplete() {
+                callback.onComplete();
+            }
+        }).execute(url.toString());
+    }
+
+    @Override
+    public void getAddress(Location location, final Callback<AddressResponse> callback) {
+        StringBuilder url =
+                new StringBuilder(Constant.GEO_CODE_API)
+                        .append("?")
+                        .append(Constant.LATLNG_PART)
+                        .append("=")
+                        .append(location.getLatitude() + "")
+                        .append(",")
+                        .append(location.getLongitude() + "");
+        new LoadDataAsync(new Callback<String>() {
+            @Override
+            public void onStartLoading() {
+                callback.onStartLoading();
+            }
+
+            @Override
+            public void onGetSuccess(String data) {
+                try {
+                    if (data == null) {
+                        callback.onGetFailure
+                                (Resources.getSystem().getString(R.string.title_load_failure));
```

Comment:
ERROR: 'method call lparen' have incorrect indentation level 32, expected level should be one of the following: 24, 28.